src/Microsoft.AspNet.TestHost/TestServer.cs

```diff
@@ -129,7 +127,7 @@ public Task Invoke(HttpContext context)
             {
                 throw new ObjectDisposedException(GetType().FullName);
             }
-            return _appDelegate(context);
+            return _requestHandler(context.Features);
```

Comment:
Combined with the comment above, we should determine what the test server should be responsible for. Originally the `ClientHandler` and `WebSocketClient` handles context creation and dispose and Invoke only handles processing. Now Invoke handles all of that and as a side effect recreates a new default HttpContext every time from the features inside the given httpcontext.